### PR TITLE
Remediate [BVR-02C, EHF-02C, PHF-02C] Redundant Parenthesis Statement

### LINKED ISSUE #733: [BVR-02C] Redundant Parenthesis Statements
## BVR-02C: Redundant Parenthesis Statements

| Type | Severity | Location |
| ---- | -------- | -------- |
| [Code Style](/appendix/finding-types#code-style) |  | [BosonVoucher.sol:L113](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/clients/voucher/BosonVoucher.sol#L113), [L223](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/clients/voucher/BosonVoucher.sol#L223), [L279](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/clients/voucher/BosonVoucher.sol#L279), [L282](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/clients/voucher/BosonVoucher.sol#L282), [L317](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/clients/voucher/BosonVoucher.sol#L317) |

### Description:

The referenced statements are redundantly wrapped in parenthesis' (`()`).

### Example:

```solidity
require((_tokenId < rangeStart) || (_tokenId >= rangeStart + range.length), EXCHANGE_ID_IN_RESERVED_RANGE);
```

### Recommendation:

We advise them to be safely omitted, increasing the legibility of the codebase.

### LINKED ISSUE #735: [EHF-02C] Redundant Parenthesis Statement
## EHF-02C: Redundant Parenthesis Statement

| Type | Severity | Location |
| ---- | -------- | -------- |
| [Code Style](/appendix/finding-types#code-style) |  | [ExchangeHandlerFacet.sol:L1065](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/facets/ExchangeHandlerFacet.sol#L1065) |

### Description:

The referenced statement is redundantly wrapped in parenthesis (`()`).

### Example:

```solidity
return (IERC721(_condition.tokenAddress).ownerOf(_tokenId) == _buyer);
```

### Recommendation:

We advise them to be safely omitted, increasing the legibility of the codebase.

### LINKED ISSUE #739: [PHF-02C] Redundant Parenthesis Statement
## PHF-02C: Redundant Parenthesis Statement

| Type | Severity | Location |
| ---- | -------- | -------- |
| [Code Style](/appendix/finding-types#code-style) |  | [PauseHandlerFacet.sol:L89](https://github.com/bosonprotocol/boson-protocol-contracts/blob/02a4d2ff04832a4607674ba37708d4f52cdac4fa/contracts/protocol/facets/PauseHandlerFacet.sol#L89) |

### Description:

The referenced statement is redundantly wrapped in parenthesis (`()`).

### Example:

```solidity
if ((status.pauseScenario & (1 << i)) != 0) {
```

### Recommendation:

We advise them to be safely omitted, increasing the legibility of the codebase.

### PR DESCRIPTION
**Pull request by zajck**
Close #733 
Close #735 
Close #739

### CODE REVIEW

**Review by anajuliabit — APPROVED** [review]

**Review by albertfolch-redeemeum — DISMISSED** [review]

File: contracts/protocol/facets/PauseHandlerFacet.sol [hunk 1]
```diff
@@ -86,7 +86,7 @@ contract PauseHandlerFacet is ProtocolBase, IBosonPauseHandler {
 
             for (uint256 i = 0; i < totalRegions; i++) {
                 // Check if the region is paused by bitwise AND operation with shifted 1
-                if ((status.pauseScenario & (1 << i)) != 0) {
+                if (status.pauseScenario & (1 << i) != 0) {
```
Comment (levalleux-ludo) [inline contracts/protocol/facets/PauseHandlerFacet.sol:89]: I personally prefer explicit statement with braces. 
However, in the logic of this PR, and according to https://docs.soliditylang.org/en/v0.8.17/cheatsheet.html#order-of-precedence-of-operators, the bitwise shift "<<" will be resolved before the bitwise AND "&".
So these remaining braces are still useless
```suggestion
                if (status.pauseScenario & 1 << i != 0) {
```

**Review by levalleux-ludo — COMMENTED** [review]
small remark

File: contracts/protocol/facets/PauseHandlerFacet.sol (hunk 1 above)
Comment (zajck) [reply contracts/protocol/facets/PauseHandlerFacet.sol:87]: You are right.
I applied the change, but the prettier put these parentheses back.

I cannot find the setting to turn it off. And I think in this place, although they are logically not needed, they actually add to the legibility of the code. 

Also, when it comes to the combination of bitwise shift and some other bitwise operator, we wrap it in parentheses in other places in the code.

So if you agree, I would just keep it as it is now.

Comment (levalleux-ludo) [reply contracts/protocol/facets/PauseHandlerFacet.sol:87]: no pb

**Review by levalleux-ludo — APPROVED** [review]

**Review by albertfolch-redeemeum — APPROVED** [review]

**Pull request merged**